### ISSUE TITLE
act does not support matrix variables with "-" in their names, GitHub does

### ISSUE THREAD

**Issue by adamtheturtle** [nektos/act#337]
I had:

```yaml
      matrix:
        python-version: [3.7, 3.8]
```

and

```yaml
    steps:
      - uses: actions/checkout@v2
      - name: "Set up Python"
        uses: actions/setup-python@v1
        with:
          python-version: ${{ matrix.python-version }}
```

This works on GitHub but `act` shows:

```
ERRO[0004] Unable to interpolate string '${{ matrix.python-version }}' - [ReferenceError: 'version' is not defined] 
```

**Comment by justincy** [comment]
Related: https://github.com/nektos/act/issues/315

**Comment by cdb** [comment]
@adamtheturtle - this looks like it was fixed just a couple days ago in `0.2.12` https://github.com/nektos/act/pull/287#issuecomment-683121952

**Comment by adamtheturtle** [comment]
Thank you.
